### PR TITLE
fix: use wait condition to rewrite datarequest

### PR DESCRIPTION
**Pull request by xiaoyifang**

### CODE REVIEW

**Comment by shenlebantongying** [comment]
github action runners secreatly has linuxbrew installed, https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#homebrew-note

Those commands will make clang-foramt 16 available.
```
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
brew install clang-format
echo "/home/linuxbrew/.linuxbrew/opt/clang-format/bin" >> $GITHUB_PATH
```

**Comment by xiaoyifang** [comment]
you have saved my life .

the clang-format in the action suddenly failed to run .

**Comment by shenlebantongying** [comment]
A typo :sweat_smile: 
Should be

```
/home/linuxbrew/.linuxbrew/opt/clang-format/bin/git-clang-format --style=file HEAD^
```

**Comment by shenlebantongying** [comment]
wtf why it failed :sweat_smile:

The version used is clang-format 16 based on my testing:
https://github.com/shenlebantongying/ci/actions/runs/5681322993/job/15397228405#step:2:163

**Comment by shenlebantongying** [comment]
I think it should be `echo "$(brew --prefix clang-format)/bin" >> "$GITHUB_PATH"`

https://github.com/search?q=eval+%22%24%28%2Fhome%2Flinuxbrew%2F.linuxbrew%2Fbin%2Fbrew+shellenv%29%22+clang-format+path%3A.github&type=code

File: src/dict/dictionary.cc [hunk 1]
```diff
@@ -116,25 +116,32 @@ void DataRequest::appendDataSlice( const void * buffer, size_t size )
...
 void DataRequest::getDataSlice( size_t offset, size_t size, void * buffer )
 {
-  if ( size == 0 )
-    return;
-
   QMutexLocker _( &dataMutex );
 
+  if ( size == 0 ) {
+    cond.wait( &dataMutex, 10 );
+    return;
```
Comment (xiaoyifang) [inline src/dict/dictionary.cc:136]: There are chances that the code will reach here.
wait on an unreached signal.

the timeout here is just a workaround.  need more investigation.

**Pull request merged**